[App] Fix TypeScript errors in orpc client and validator page

## Context

TypeScript errors detected in `packages/app` during CI type-check.

## Errors

### 1. Missing module declaration
```
lib/orpc.ts(3,29): error TS2307: Cannot find module '@beacon-indexer/api/routers' or its corresponding type declarations.
```

### 2. Type errors in validator page
```
app/validator/[id]/page.tsx(29,35): error TS2339: Property 'success' does not exist on type '{}'.
app/validator/[id]/page.tsx(29,54): error TS2339: Property 'data' does not exist on type '{}'.
app/validator/[id]/page.tsx(30,42): error TS2339: Property 'success' does not exist on type '{}'.
app/validator/[id]/page.tsx(30,61): error TS2339: Property 'error' does not exist on type '{}'.
```

## Required Changes

- [ ] Ensure `@beacon-indexer/api` exports routers types correctly
- [ ] Fix type definitions for the oRPC client response in validator page

## Notes

These are pre-existing errors discovered during #92 CI run.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -46,6 +46,9 @@ jobs:
       - name: Build db package
         run: pnpm --filter @beacon-indexer/db run build
 
+      - name: Build api package
+        run: pnpm --filter @beacon-indexer/api run build
+
       - name: Type check
         run: pnpm run type-check
 

